Products/CMFPlone/browser/favicon.py

```diff
@@ -18,9 +20,16 @@ def __init__(self, context, request):
         settings = registry.forInterface(ISiteSchema, prefix="plone")
         if getattr(settings, 'site_favicon', False):
             filename, data = b64decode_file(settings.site_favicon)
-            data = NamedImage(data=data, filename=filename)
+            mimetype = getattr(settings, 'site_favicon_mimetype', "image/vnd.microsoft.icon")
+            data = NamedImage(data=data, contentType=mimetype, filename=filename)
             self.data = data
             self.filename = filename
+        else:
+            basedir = os.path.dirname(os.path.dirname(__file__))
+            with open(os.path.join(basedir, "skins", "plone_images", "favicon.ico"), "rb") as icon:
```

Comment:
How fortunate that we ship the icon ourselves!

Reply:
Had to rename the icon, otherwise the skin would trump the view.  That's fine tho — we have a view, so the skin element being absent is no longer a problem anymore.